server/src/main/java/io/crate/analyze/AnalyzedAlterTableAddColumn.java

```diff
@@ -80,6 +86,13 @@ public AddColumnRequest bind(NodeContext nodeCtx,
                 }
             }
             newColumns.add(reference);
+            if (reference instanceof IndexReference indexRef) {
+                String analyzer = indexRef.analyzer();
+                if (fulltextAnalyzerResolver.hasCustomAnalyzer(analyzer)) {
+                    Settings settings = fulltextAnalyzerResolver.resolveFullCustomAnalyzerSettings(analyzer);
+                    builder.put(settings);
+                }
+            }
         }
```

Comment:
Would it be possible to do this in the transport action on the master node instead?

Keeping the request data closer to what's in the SQL statement has the advantage that we're more flexible in regards to changes as we don't have to make streaming changes and `settings` in particular has the downside that it is rather vague in regards to what exactly can be in it.

We should already be streaming the indexReference if I'm not mistaken, which means the other side already has all information needed and should be able to resolve the custom analyzer settings too.

Reply:
The issue with constructing it in the action (using the List<Reference> already in the request) is that we need `fulltextAnalyzerResolver` to check for custom analyzer, and to get the `Settings` needed to update the `indexAnalyzers` in `IndexService`.